src/pkg/repository/repository_test.go

```diff
@@ -284,13 +284,88 @@ func (suite *RepositoryIntegrationSuite) TestConnect_DisableMetrics() {
 	// need to initialize the repository before we can test connecting to it.
 	st := tester.NewPrefixedS3Storage(t)
 
-	repo, err := repository.Initialize(ctx, account.Account{}, st, control.Defaults())
+	repo, err := Initialize(ctx, account.Account{}, st, control.Defaults())
 	require.NoError(t, err)
 
 	// now re-connect
-	r, err := repository.Connect(ctx, account.Account{}, st, repo.GetID(), control.Options{DisableMetrics: true})
+	r, err := Connect(ctx, account.Account{}, st, repo.GetID(), control.Options{DisableMetrics: true})
 	assert.NoError(t, err)
 
 	// now we have repoID beforehand
 	assert.Equal(t, r.GetID(), r.GetID())
 }
+
+// Test_Options tests that the options are passed through to the repository
+// correctly
+func (suite *RepositoryIntegrationSuite) Test_Options() {
+	table := []struct {
+		name       string
+		opts       func() control.Options
+		assertFunc func(t *testing.T, opts control.Options)
```

Comment:
since you're only using `len` checks here, this could be simplified to an `expectedLen` int instead.